src/metric-engine/src/engine/create.rs

```diff
@@ -323,6 +335,39 @@ impl MetricEngineInner {
             ConflictRegionOptionSnafu {}
         );
 
+        // check if only one field column is declared, and all tag columns are string
+        let mut field_col: Option<&ColumnMetadata> = None;
+        for col in &request.column_metadatas {
+            if col.semantic_type == SemanticType::Field {
+                if field_col.is_some() {
+                    MultipleFieldColumnSnafu {
+                        previous: field_col.unwrap().column_schema.name.clone(),
+                        current: col.column_schema.name.clone(),
+                    }
+                    .fail()?;
+                }
+                field_col = Some(col);
+            } else if col.semantic_type == SemanticType::Tag {
+                ensure!(
+                    col.column_schema.data_type == ConcreteDataType::string_datatype(),
+                    ColumnTypeMismatchSnafu {
+                        expect: ConcreteDataType::string_datatype(),
+                        actual: col.column_schema.data_type.clone(),
+                    }
+                );
+            }
+        }
+        let field_col = field_col.context(NoFieldColumnSnafu)?;
+
+        // make sure the field column is float64 type
```

Comment:
Looks like we don't need to add this restriction. The all numeric type must be allowed.

Reply:
The key point here is every logical table's field column should have the same type.